tools/secret-manager-helper/src/main/java/com/google/cloud/secrethelper/SecretManagerStringSubstitutor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.secrethelper;
+
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.text.StringSubstitutor;
+import org.apache.commons.text.lookup.StringLookup;
+import org.apache.commons.text.lookup.StringLookupFactory;
+
+/**
+ * A class that provides a {@link org.apache.commons.text.StringSubstitutor}. It replaces template
+ * strings with either a Secret Manager payload or writes the payload to a file, and substitutes the
+ * path to the file.
+ */
+public class SecretManagerStringSubstitutor extends StringSubstitutor {
+
+  public SecretManagerStringSubstitutor(SecretManagerServiceClient client) {
+    super(
+        StringLookupFactory.INSTANCE.interpolatorStringLookup(
+            ImmutableMap.<String, StringLookup>builder()
+                .put(
+                    "secretManager",
+                    StringLookupFactory.INSTANCE.interpolatorStringLookup(new SecretLookup(client)))
+                .put(
+                    "secretManagerToFilePath",
+                    StringLookupFactory.INSTANCE.interpolatorStringLookup(
+                        new SecretLookupToFile(client)))
+                .build(),
+            null,
+            true));
```

Comment:
This is confusing. I think it would be better to create each one of the arguments passed to `super` as local variables, and then use those local variables in the call to `super`.

Reply:
Thanks for the comment. Agree it could be simplified, but: 

```
Call to 'super()' must be first statement in constructor body
```

I can simplify this by not using the builder:

```
    super(
        StringLookupFactory.INSTANCE.interpolatorStringLookup(ImmutableMap.of(
            "secretManager", StringLookupFactory.INSTANCE.interpolatorStringLookup(new SecretLookup(client)),
            "secretManagerToFilePath", StringLookupFactory.INSTANCE.interpolatorStringLookup(
                new SecretLookupToFile(client)))));
  }
```

Reasonable?

Reply:
Ah, right, apologies. Yes, that's clearer. Thanks.

Reply:
Great, thanks. I pushed the change in 9ac582c5809a76aa4840dc599159ff49fb9662cc.